src/chat/Chat.js

```diff
@@ -9,12 +9,28 @@ import MessageList from '../message-list/MessageList';
 import NoMessages from '../message/NoMessages';
 import ComposeBox from '../compose/ComposeBox';
 
-export default class MainScreen extends React.Component {
+export default class Chat extends React.Component {
+
+  unsentMessageIds = [];
+  lastSentTime = 0;
+
+  markAsRead = (messageIds) => {
+    const { auth, markMessagesRead, updateMessageFlags } = this.props;
+
+    this.unsentMessageIds.push(...messageIds);
+    markMessagesRead(this.unsentMessageIds);
```

Comment:
I think this will slow performance considerably while scrolling fast through unread messages. We shouldn't be calling actions this frequently. That's why I originally had it mutate the message state